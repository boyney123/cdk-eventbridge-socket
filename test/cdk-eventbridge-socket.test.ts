import {
  SynthUtils,
  expect as expectCDK,
  haveResourceLike,
} from '@aws-cdk/assert';
import { Stack } from '@aws-cdk/core';
import { EventBridgeWebSocket } from '../lib';

describe('EventBridgeWebSocket', () => {
  it('snapshot test EventsRuleToSns default params', () => {
    const stack = new Stack();
    new EventBridgeWebSocket(stack, 'eventBridgeSocketDeploy', {
      bus: 'my-random-bus',
      eventPattern: {
        account: ['my-account'],
      },
      stage: 'dev',
    });
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });

  it('creates an API Gateway WEBSOCKET with onconnect and ondisconnect integrations', () => {
    const stack = new Stack();

    new EventBridgeWebSocket(stack, 'eventBridgeSocketDeploy', {
      bus: 'my-random-bus',
      eventPattern: {
        account: ['my-account'],
      },
      stage: 'dev',
    });

    // console.log(JSON.stringify(SynthUtils.toCloudFormation(stack), null, 4));
    expectCDK(stack).to(
      haveResourceLike('AWS::ApiGatewayV2::Api', {
        Name: 'EventBridgeSockets',
        ProtocolType: 'WEBSOCKET',
        RouteSelectionExpression: '$request.body.action',
      })
    );

    // on connect integration
    expectCDK(stack).to(
      haveResourceLike('AWS::ApiGatewayV2::Integration', {
        ApiId: {
          Ref: 'eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47',
        },
        IntegrationType: 'AWS_PROXY',
        IntegrationUri: {
          'Fn::Join': [
            '',
            [
              'arn:',
              {
                Ref: 'AWS::Partition',
              },
              ':apigateway:',
              {
                Ref: 'AWS::Region',
              },
              ':lambda:path/2015-03-31/functions/',
              {
                'Fn::GetAtt': [
                  'eventBridgeSocketDeployonconnectAE0ACD17',
                  'Arn',
                ],
              },
              '/invocations',
            ],
          ],
        },
      })
    );

    // expectCDK(stack).to(
    haveResourceLike('AWS::ApiGatewayV2::Route', {
      ApiId: {
        Ref: 'eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47',
      },
      RouteKey: '$connect',
      Target: {
        'Fn::Join': [
          '',
          [
            'integrations/',
            {
              Ref: 'eventBridgeSocketDeployeventBridgeSocketDeployapiconnectRouteWebSocketIntegration449f455cb3c7748381eb0f0246ccfa2f05649A15',
            },
          ],
        ],
      },
    });

    expectCDK(stack).to(
      haveResourceLike('AWS::ApiGatewayV2::Integration', {
        ApiId: {
          Ref: 'eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47',
        },
        IntegrationType: 'AWS_PROXY',
        IntegrationUri: {
          'Fn::Join': [
            '',
            [
              'arn:',
              {
                Ref: 'AWS::Partition',
              },
              ':apigateway:',
              {
                Ref: 'AWS::Region',
              },
              ':lambda:path/2015-03-31/functions/',
              {
                'Fn::GetAtt': [
                  'eventBridgeSocketDeployondisconnect0F61A161',
                  'Arn',
                ],
              },
              '/invocations',
            ],
          ],
        },
      })
    );

    expectCDK(stack).to(
      haveResourceLike('AWS::ApiGatewayV2::Route', {
        ApiId: {
          Ref: 'eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47',
        },
        RouteKey: '$disconnect',
        Target: {
          'Fn::Join': [
            '',
            [
              'integrations/',
              {
                Ref: 'eventBridgeSocketDeployeventBridgeSocketDeployapidisconnectRouteWebSocketIntegration0a84f9265d12d31c9984b952fd3b77202D79D90F',
              },
            ],
          ],
        },
      })
    );
  });

  it('creates a new EventBridge rule for given event bus and given event pattern', () => {
    const stack = new Stack();
    new EventBridgeWebSocket(stack, 'eventBridgeSocketDeploy', {
      bus: 'my-random-bus',
      eventPattern: {
        account: ['my-account'],
      },
      stage: 'dev',
    });

    expectCDK(stack).to(
      haveResourceLike('AWS::Events::Rule', {
        EventBusName: 'my-random-bus',
        EventPattern: {
          account: ['my-account'],
        },
      })
    );
  });
});
