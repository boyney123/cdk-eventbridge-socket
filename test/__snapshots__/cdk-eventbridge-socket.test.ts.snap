// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventBridgeWebSocket snapshot test EventsRuleToSns default params 1`] = `
Object {
  "Outputs": Object {
    "eventBridgeSocketDeployWebsocketendpoint7B865CDA": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Fn::GetAtt": Array [
                "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
                "ApiEndpoint",
              ],
            },
            "/dev",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9ArtifactHash8FE9313B": Object {
      "Description": "Artifact hash for asset \\"0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9\\"",
      "Type": "String",
    },
    "AssetParameters0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9S3BucketF5637FC3": Object {
      "Description": "S3 bucket for asset \\"0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9\\"",
      "Type": "String",
    },
    "AssetParameters0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9S3VersionKey9EDF8659": Object {
      "Description": "S3 key for asset version \\"0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9\\"",
      "Type": "String",
    },
    "AssetParameters0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cefArtifactHash0CEE3A9F": Object {
      "Description": "Artifact hash for asset \\"0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cef\\"",
      "Type": "String",
    },
    "AssetParameters0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cefS3Bucket0CDABFD2": Object {
      "Description": "S3 bucket for asset \\"0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cef\\"",
      "Type": "String",
    },
    "AssetParameters0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cefS3VersionKeyFDD215FC": Object {
      "Description": "S3 key for asset version \\"0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cef\\"",
      "Type": "String",
    },
    "AssetParameters4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4ArtifactHashD218146F": Object {
      "Description": "Artifact hash for asset \\"4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4\\"",
      "Type": "String",
    },
    "AssetParameters4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4S3Bucket6200EC24": Object {
      "Description": "S3 bucket for asset \\"4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4\\"",
      "Type": "String",
    },
    "AssetParameters4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4S3VersionKeyB7169A21": Object {
      "Description": "S3 key for asset version \\"4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "eventBridgeSocketDeployWebsocketRule71C1F191": Object {
      "Properties": Object {
        "EventBusName": "my-random-bus",
        "EventPattern": Object {
          "account": Array [
            "my-account",
          ],
        },
        "State": "ENABLED",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "eventBridgeSocketDeployeventbridgebroker1D8651C6",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "eventBridgeSocketDeployWebsocketRuleAllowEventRuleeventBridgeSocketDeployeventbridgebroker9CB272C85390421F": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "eventBridgeSocketDeployeventbridgebroker1D8651C6",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "eventBridgeSocketDeployWebsocketRule71C1F191",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47": Object {
      "Properties": Object {
        "Name": "EventBridgeSockets",
        "ProtocolType": "WEBSOCKET",
        "RouteSelectionExpression": "$request.body.action",
      },
      "Type": "AWS::ApiGatewayV2::Api",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapiconnectRoute308BAAC5": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
        },
        "RouteKey": "$connect",
        "Target": Object {
          "Fn::Join": Array [
            "",
            Array [
              "integrations/",
              Object {
                "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiconnectRouteWebSocketIntegration449f455cb3c7748381eb0f0246ccfa2f05649A15",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapiconnectRouteWebSocketIntegration449f455cb3c7748381eb0f0246ccfa2f05649A15": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":apigateway:",
              Object {
                "Ref": "AWS::Region",
              },
              ":lambda:path/2015-03-31/functions/",
              Object {
                "Fn::GetAtt": Array [
                  "eventBridgeSocketDeployonconnectAE0ACD17",
                  "Arn",
                ],
              },
              "/invocations",
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapiconnectRouteeventBridgeSocketDeployeventBridgeSocketDeployapiconnectRoute969E1C59PermissionE0DAFB7C": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "eventBridgeSocketDeployonconnectAE0ACD17",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
              },
              "/*/*$connect",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapidisconnectRoute5AA2FBC6": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
        },
        "RouteKey": "$disconnect",
        "Target": Object {
          "Fn::Join": Array [
            "",
            Array [
              "integrations/",
              Object {
                "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapidisconnectRouteWebSocketIntegration0a84f9265d12d31c9984b952fd3b77202D79D90F",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapidisconnectRouteWebSocketIntegration0a84f9265d12d31c9984b952fd3b77202D79D90F": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":apigateway:",
              Object {
                "Ref": "AWS::Region",
              },
              ":lambda:path/2015-03-31/functions/",
              Object {
                "Fn::GetAtt": Array [
                  "eventBridgeSocketDeployondisconnect0F61A161",
                  "Arn",
                ],
              },
              "/invocations",
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapidisconnectRouteeventBridgeSocketDeployeventBridgeSocketDeployapidisconnectRoute1F6837D8Permission67DA007B": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "eventBridgeSocketDeployondisconnect0F61A161",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
              },
              "/*/*$disconnect",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployapistage6E463CF6": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
        },
        "AutoDeploy": true,
        "StageName": "dev",
      },
      "Type": "AWS::ApiGatewayV2::Stage",
    },
    "eventBridgeSocketDeployeventBridgeSocketDeployconnectionstable5C7E957A": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "connectionId",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "connectionId",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "TableName": "eventBridgeSocketDeploy-connections-table",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
    "eventBridgeSocketDeployeventbridgebroker1D8651C6": Object {
      "DependsOn": Array [
        "eventBridgeSocketDeployeventbridgebrokerServiceRoleDefaultPolicy5D548830",
        "eventBridgeSocketDeployeventbridgebrokerServiceRoleE80F517C",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9S3BucketF5637FC3",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9S3VersionKey9EDF8659",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0465dbaf624c7c75707a7059f7f4d57a99144d52abf74b0056954cb5c4f730f9S3VersionKey9EDF8659",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "TABLE_NAME": "eventBridgeSocketDeploy-connections-table",
            "WEBSOCKET_API": Object {
              "Fn::Join": Array [
                "",
                Array [
                  Object {
                    "Fn::GetAtt": Array [
                      "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
                      "ApiEndpoint",
                    ],
                  },
                  "/dev",
                ],
              ],
            },
          },
        },
        "Handler": "eventbridge-broker.handler",
        "MemorySize": 256,
        "Role": Object {
          "Fn::GetAtt": Array [
            "eventBridgeSocketDeployeventbridgebrokerServiceRoleE80F517C",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "eventBridgeSocketDeployeventbridgebrokerServiceRoleDefaultPolicy5D548830": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "execute-api:ManageConnections",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:execute-api:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":",
                    Object {
                      "Ref": "eventBridgeSocketDeployeventBridgeSocketDeployapiA13E2D47",
                    },
                    "/*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "eventBridgeSocketDeployeventBridgeSocketDeployconnectionstable5C7E957A",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "eventBridgeSocketDeployeventbridgebrokerServiceRoleDefaultPolicy5D548830",
        "Roles": Array [
          Object {
            "Ref": "eventBridgeSocketDeployeventbridgebrokerServiceRoleE80F517C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "eventBridgeSocketDeployeventbridgebrokerServiceRoleE80F517C": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "eventBridgeSocketDeployonconnectAE0ACD17": Object {
      "DependsOn": Array [
        "eventBridgeSocketDeployonconnectServiceRoleDefaultPolicy4A968F2B",
        "eventBridgeSocketDeployonconnectServiceRole1FF5024C",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4S3Bucket6200EC24",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4S3VersionKeyB7169A21",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4093a22ea6ccf39bbe5a10a1aaff656ed40dce829d913111c9b1a37877f1f8a4S3VersionKeyB7169A21",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "TABLE_NAME": "eventBridgeSocketDeploy-connections-table",
          },
        },
        "Handler": "on-connect.handler",
        "MemorySize": 256,
        "Role": Object {
          "Fn::GetAtt": Array [
            "eventBridgeSocketDeployonconnectServiceRole1FF5024C",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "eventBridgeSocketDeployonconnectServiceRole1FF5024C": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "eventBridgeSocketDeployonconnectServiceRoleDefaultPolicy4A968F2B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "eventBridgeSocketDeployeventBridgeSocketDeployconnectionstable5C7E957A",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "eventBridgeSocketDeployonconnectServiceRoleDefaultPolicy4A968F2B",
        "Roles": Array [
          Object {
            "Ref": "eventBridgeSocketDeployonconnectServiceRole1FF5024C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "eventBridgeSocketDeployondisconnect0F61A161": Object {
      "DependsOn": Array [
        "eventBridgeSocketDeployondisconnectServiceRoleDefaultPolicyC9FDF75A",
        "eventBridgeSocketDeployondisconnectServiceRoleA5D3C57B",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cefS3Bucket0CDABFD2",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cefS3VersionKeyFDD215FC",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0bffb408a9c50735b5bd38bd55f84a27977a0ee2f1506ba94759f266593d9cefS3VersionKeyFDD215FC",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "TABLE_NAME": "eventBridgeSocketDeploy-connections-table",
          },
        },
        "Handler": "on-disconnect.handler",
        "MemorySize": 256,
        "Role": Object {
          "Fn::GetAtt": Array [
            "eventBridgeSocketDeployondisconnectServiceRoleA5D3C57B",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "eventBridgeSocketDeployondisconnectServiceRoleA5D3C57B": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "eventBridgeSocketDeployondisconnectServiceRoleDefaultPolicyC9FDF75A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "eventBridgeSocketDeployeventBridgeSocketDeployconnectionstable5C7E957A",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "eventBridgeSocketDeployondisconnectServiceRoleDefaultPolicyC9FDF75A",
        "Roles": Array [
          Object {
            "Ref": "eventBridgeSocketDeployondisconnectServiceRoleA5D3C57B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
